src/logger/logger.hpp

```diff
@@ -354,12 +355,19 @@
   do {                                                                \
     auto throw_error = [&]() GL_COLD_NOINLINE_ERROR {    \
       logstream(LOG_ERROR) << (message) << std::endl;                 \
-      throw(std::ios_base::failure(message));                         \
+      throw(turi::error::io_error(message));                         \
     };                                                                \
     throw_error();                                                    \
   } while(0)
 #endif
 
+#define log_and_throw_current_io_failure()                            \
+  do {                                                                \
+    auto error_code = errno;                                          \
+    std::string error_message = std::strerror(error_code);            \
```

Comment:
We should clear errno with `errno = 0;` after "handling" the error (here handling is done by raising an exception).

Reply:
Done.